rpcs3/Emu/Io/LogitechG27Config.h

```diff
@@ -20,15 +20,61 @@ enum class hat_component
 	right
 };
 
+// this was a bitfield, but juggling at least 3 compilers and OSes means no bitfield
+// num_buttons:10 << 52 | num_hats:10 << 42 | num_axes:10 << 32 | vendor_id:16 << 16 | product_id:16
+struct emulated_g27_device_type_id
+{
+	// big types to keep 64bit bit shift operations sane
+	u64 product_id;
+	u64 vendor_id;
+	u64 num_axes;
+	u64 num_hats;
+	u64 num_buttons;
+
+	u64 as_u64() const
+	{
+		u64 value = product_id;
+		value |= vendor_id << 16;
+		value |= (num_axes & ((1 << 10) - 1)) << 32;
+		value |= (num_hats & ((1 << 10) - 1)) << 42;
+		value |= (num_buttons & ((1 << 10) - 1)) << 52;
+		return value;
+	}
+
+	bool is_v1()
+	{
+		return !num_axes && !num_hats && !num_buttons;
+	}
+
+	static emulated_g27_device_type_id from_u64(u64 data)
+	{
+		const emulated_g27_device_type_id id =
+		{
+			.product_id = static_cast<u64>(data & 0xFFFF),
+			.vendor_id = static_cast<u64>((data >> 16) & 0xFFFF),
+			.num_axes = static_cast<u64>((data >> 32) & ((1 << 10) - 1)),
+			.num_hats = static_cast<u64>((data >> 42) & ((1 << 10) - 1)),
+			.num_buttons = static_cast<u64>((data >> 52) & ((1 << 10) - 1))
+		};
+		return id;
+	}
+
+	static bool is_v1(u64 data)
+	{
+		emulated_g27_device_type_id id = from_u64(data);
+		return id.is_v1();
+	}
+};
+
 struct emulated_logitech_g27_mapping : cfg::node
 {
-	cfg::uint<0, 0xFFFFFFFF> device_type_id;
+	cfg::uint<0, 0xFFFFFFFFFFFFFFFF> device_type_id;
```

Comment:
Can't you just use the GUID string and save that instead ?
See #17209

Reply:
With one wheel twin devices from Fanatec, I'd rather not trust the GUID generation for now

https://github.com/libsdl-org/SDL/blob/1eeffc5933d49ee3b9de032a205da530ee83c786/src/joystick/windows/SDL_dinputjoystick.c#L459-L548

https://github.com/RPCS3/rpcs3/issues/17191#issuecomment-2874755411